src/main/scala/is/hail/table/Table.scala

```diff
@@ -47,8 +47,10 @@ case class TableSpec(
 }
 
 object Table {
-  def range(hc: HailContext, n: Int, nPartitions: Option[Int] = None): Table =
-    new Table(hc, TableRange(n, nPartitions.getOrElse(hc.sc.defaultParallelism)))
+  def range(hc: HailContext, n: Int, nPartitions: Option[Int] = None): Table = {
+    require(nPartitions.forall(_ <= n))
```

Comment:
I think it would be better to just take the `min(n, nPartitions)` if nPartitions is defined. Otherwise, this should be a check in the Python side with a nice error message.

Reply:
I'll just move the `min` into the IR like `MatrixRange` does it. I was hesitant to make the IR node produce a different number of partitions than requested, but Cotton doesn't have a problem with it.